src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -872,53 +872,82 @@ private void BeforeAppendToRaftLog(ClusterOperationContext ctx, CommandBase cmd)
-                        if (addDatabase.Record.Sharding.BucketRanges == null ||
-                            addDatabase.Record.Sharding.BucketRanges.Count == 0)
-                        {
-                            addDatabase.Record.Sharding.BucketRanges = new List<ShardBucketRange>();
-                            var start = 0;
-                            var step = ShardHelper.NumberOfBuckets / addDatabase.Record.Sharding.Shards.Length;
-                            for (int i = 0; i < addDatabase.Record.Sharding.Shards.Length; i++)
-                            {
-                                addDatabase.Record.Sharding.BucketRanges.Add(new ShardBucketRange
-                                {
-                                    ShardNumber = i,
-                                    BucketRangeStart = start
-                                });
-                                start += step;
-                            }
-                        }
+                        FillShardingConfiguration(addDatabase, clusterTopology);
+                    }
+                    break;
+            }
+        }
 
-                        var orchestratorTopology = addDatabase.Record.Sharding.Orchestrator.Topology;
-                        if (orchestratorTopology.Count == 0)
-                            AssignNodesToDatabase(clusterTopology, addDatabase.Record.DatabaseName, addDatabase.Encrypted, orchestratorTopology);
+        private void FillShardingConfiguration(AddDatabaseCommand addDatabase, ClusterTopology clusterTopology)
+        {
+            var shardingConfiguration = addDatabase.Record.Sharding;
+            if (shardingConfiguration.BucketRanges == null ||
+                shardingConfiguration.BucketRanges.Count == 0)
+            {
+                shardingConfiguration.BucketRanges = new List<ShardBucketRange>();
+                var start = 0;
+                var step = ShardHelper.NumberOfBuckets / shardingConfiguration.Shards.Length;
+                for (int i = 0; i < shardingConfiguration.Shards.Length; i++)
+                {
+                    shardingConfiguration.BucketRanges.Add(new ShardBucketRange
+                    {
+                        ShardNumber = i, 
+                        BucketRangeStart = start
+                    });
+                    start += step;
+                }
+            }
 
-                        Debug.Assert(orchestratorTopology.Count != 0, "Empty orchestrator topology after AssignNodesToDatabase");
+            if (shardingConfiguration.Prefixed != null)
+            {
+                var prefixedSettings = shardingConfiguration.Prefixed.Values.ToList();
 
-                        var pool = GetNodesDistribution(clusterTopology, addDatabase.Record.Sharding.Shards);
+                for (int i = 0; i < prefixedSettings.Count; i++)
+                {
+                    var shards = prefixedSettings[i].Shards;
+                    var start = ShardHelper.NumberOfBuckets << i;
```

Comment:
This multiples, not add.
So you have `0.. 1M, 1M .. 2M, 4M..5M, 8M..9M`, no?

Reply:
fixed